vm/src/dictdatatype.rs

```diff
@@ -65,7 +69,9 @@ struct DictInner<T> {
     used: usize,
     filled: usize,
     indices: Vec<i64>,
-    entries: Vec<DictEntry<T>>,
+    entries: Vec<Option<DictEntry<T>>>,
+    // index to new inserted element should be in entries
+    nentries: usize,
```

Comment:
```suggestion
    next_new_entry_idx: usize,
```
I found `nentries` is used in different meaning from CPython.

Reply:
I thought it has similar work in pop_back. i would fix that.